addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3537,6 +3545,10 @@ const SnippetOptionWidget = Widget.extend({
      * @returns {Promise<string|undefined>|string|undefined}
      */
     _computeWidgetState: async function (methodName, params) {
+        const $target = params.target ? this.options.wysiwyg.getEditable().find(params.target) : this.$target;
+        if (!$target[0]) {
+            return;
+        }
```

Comment:
@Zinston You copy/paste that code in multiple generic methods, meaning their override would have to do the same, meaning that apparently-generic feature is only available to a bunch of generic methods, meaning you have to modify `this.$target` into `$target` in the whole implementation (breaking history)... That also complexifies the implementation of the method with a new "do I have the right `$target`?" at the beginning, things you should not have to care about when implementing an option method behavior (`selectStyle` = you want to force some style on the `$target`... that's it, not first asking yourself if you have the right `$target`).
Don't you think that option should have been coded on the caller side? That's what we do for the `apply-to` parameter... which is a near-duplicate of your new `target` parameter (one search in the whole DOM, one search in the original `$target`).

The minimum for me would to code it the same way as `apply-to` with a similar name. But what I think should be done is using the `apply-to` parameter as it is currently implemented and review the `this.$target` of your specific mass mailing option to be the whole editable area (maybe just a call to `setTarget` in the `init` would be enough?). Your new semi-generic parameter encourages a bad practice of having widgets displayed for a section A but changing a section B which could be anywhere in the page, far away from A => that's not the case for your specific option, the more the reason to code that specifically for your option by changing the base `$target` of your "Design" option as I suggest.

If you agree, could you make this a priority? I have small conflicts because of this, would be easier you change this first (otherwise I will be the one sending you conflicts by merging my resolved ones ^^).